crates/ra_hir/src/expr.rs

```diff
@@ -616,14 +641,110 @@ impl ExprCollector {
                 let op = e.op();
                 self.alloc_expr(Expr::BinaryOp { lhs, rhs, op }, syntax_ptr)
             }
+            ast::ExprKind::Literal(e) => {
+                let child = e.syntax().children().next();
```

Comment:
instead of using the raw syntax here, I think it would be better to use the generated AST. You'll need to extend it -- I think you need (in `grammar.ron`) to define an `enum` that can contain the various kinds of literal, and add that as an `option` to `Literal`. And for the various literal types (like `ast::String`), we'll need accessors that return the data.

Reply:
Done!